### ISSUE TITLE
Username is 0 in IRC/WOL protocol

### ISSUE THREAD

**Issue by SoraBytes** [pvpgn/pvpgn-server#461]
I've compiled PvPGN on both docker (Debian/Ubuntu/Alpine from Develop, and a known good commit), and manually (Archlinux from Develop, and a known good commit) but for some odd reason, the IRC/WOL protocol seems to be broken

I've tested against g++ and clang, Arm, and x86_64.

Basicallly usenames show up as "0" if you join the channel last. If the user who is '0' leaves and rejoins the channel it shows three names. "User1, User2, 0" The system logs are showing no symptoms, and are using the proper names. I thought perhaps it was related to WOL, so we tried IRC clients on PvPGN, but its the same issue.... IRC clients show "User1User2" in the same name.

Is this by chance a newer library issue?

Notes:
1) This occuring also limits text chat e.g. clients cannot communicate.
2) Example images are from Red Alert 2

![screencap_2021_08_18_03_24_42_AM](https://user-images.githubusercontent.com/18122133/129856227-59e7b28d-0642-4380-8c73-0fccb0676856.png)
![screencap_2021_08_18_03_25_40_AM](https://user-images.githubusercontent.com/18122133/129856251-230d89fe-6149-4ebc-8ff1-ee7b85b672de.png)

**Comment by SoraBytes** [comment]
The log file. This log doesn't match the images above, as its just myself running two clients locally, but they have the same issue.

[bnetd.log](https://github.com/pvpgn/pvpgn-server/files/7011151/bnetd.log)

**Comment by SoraBytes** [comment]
The following patch file fixes the following issues #433 #197 and #461.
I have tested this a few times with Red Alert 2, Renegade. HexChat.
I haven't encountered any issues using this patch so far.

IRC:
The IRC username tree for channels being "User1User2User3" all in one name, has been fixed.
WOL:
Renegade no longer crashes during login.
Renegade can see Dedicated Servers & Join
Chat channels no longer show "0, User1, User2" and prevents clients from communicating.

[WOL Fixes.zip](https://github.com/pvpgn/pvpgn-server/files/7042997/WOL.Fixes.zip)
